### PR TITLE
Suggestions on modules page

### PR DESCRIPTION
**Pull request by ilayaperumalg**

### CODE REVIEW

File: spring-xd-modules.tex [hunk 1]
```diff
@@ -1,44 +1,45 @@
...
 \label{sec:Modules}
-A Spring XD Module \cite{modules} is a unit of data processing and as mentioned above 
+A Spring XD Module \cite{modules} is a data processing unit and as mentioned above
 there are four types: source, processor, sink, and job. Modules in Spring XD are defined 
 in their own application context. This allows for easy encapsulation and life cycle 
 management for modules. Additionally, the use of an application context allows for easy 
 module expansion.  Spring XD uses Spring Integration \cite{spring-integration-reference} 
-as its foundation for implementing modules. A module is comprised of one or more channels 
-and an integration bean.
+as its foundation for implementing modules. A module is comprised of component that
```
Comment (pperalta) [inline spring-xd-modules.tex]: components

File: spring-xd-modules.tex [hunk 2]
```diff
@@ -1,44 +1,45 @@
...
+implements the data processing logic and one or more connectors(known as channels)
```
Comment (pperalta) [inline spring-xd-modules.tex]: implement

Comment (pperalta) [inline spring-xd-modules.tex]: space between connectors and (

File: spring-xd-modules.tex [hunk 3]
```diff
@@ -1,44 +1,45 @@
...
 
 Spring XD offers a suite of 23 sources, 24 sinks, 9 processors and 9 jobs that are ready 
-to use at startup.  These modules integrate with a variety of well
-known and popular data stores and processing systems such as JDBC, HDFS, MongoDB, Spark,
-and Sqoop.  If an existing module does not meet the needs of a given use case, Spring XD
-supports custom modules.
+to use at startup.  These modules integrate with a variety of well known and popular
+data stores and processing systems such as JDBC, HDFS, MongoDB, Spark, Kafka, RabbitMQ,
+Sqoop etc.,  If an existing module does not meet the needs of a given use case, Spring XD
+supports developing custom modules.
```
Comment (pperalta) [inline spring-xd-modules.tex]: Isn't it implied that custom modules require development? I'm not sure the addition of "developing" is needed here.

File: spring-xd-modules.tex [hunk 4]
```diff
@@ -1,44 +1,45 @@
...
 
 Spring XD sink and source modules are Message Endpoints 
 \cite{enterprise-integration-pattern-message-endpoint} 
-that are responsible for sending data to and receiving data from external applications.
-A source is the entry point for data into the stream. A sink is the module that dispatches
-the stream's results to an external application or storage system. Batch jobs are used to
-execute batch processing steps on a set of data.
+that are responsible for sending data to and receiving data from external applications
+respectively. A source is the entry point for data into the stream. A sink is the module that dispatches
+the stream's results to an external application or storage system. A Processor
```
Comment (pperalta) [inline spring-xd-modules.tex]: s/Processor/processor

File: spring-xd-modules.tex [hunk 5]
```diff
@@ -52,13 +53,14 @@ \subsection{Source}
...
 \subsection{Processor}
-A Processor module is used to modify the data as it passes on its way from the source to
-the sink.  The basic processor is comprised of both an ``input'' and an ``output'' channel
-and a bean.
-The input channel receives all messages from the stream and dispatches
-messages to the integration bean (see figure~\ref{fig:processormbc}.) It is the responsibility of
-the bean to modify the data. The modified message is then sent to next module 
-via the output channel.  
+A processor is the module that receives data from a source or a previous processor
+module's output and performs the transformation operation and sends the data
```
Comment (pperalta) [inline spring-xd-modules.tex]: module's output, performs the transformation operation and send the data

File: spring-xd-modules.tex [hunk 6]
```diff
@@ -52,13 +53,14 @@ \subsection{Source}
...
+into a sink module or a downstream processor module. The basic processor is
+comprised of both an ``input'' and an ``output'' connector channels and the data processing component.
```
Comment (pperalta) [inline spring-xd-modules.tex]: remove "an" between "both" and "input"

File: spring-xd-modules.tex [hunk 7]
```diff
@@ -72,20 +74,19 @@ \subsection{Processor}
...
 \subsection{Sink}
-Sink modules convert and deliver messages from a stream in a format consumable by 
-an external application.  There are two types of sinks: counter/gauge and delegate.
-A counter/gauge is a specialized sink that increments a field on a data store every time a
-message is received. (See section~\ref{sec:Analytics}.) A delegate
-sink translates messages to the format expected by the external application.
-After transforming the message, the resulting data is sent to the external application.
+Sink modules convert and deliver data out of the stream in a format consumable by
+an external application.  There are two types of sinks: analytic and delegate.
+An analytic sink is used to perform analytic operations such as count, gauge on the
```
Comment (pperalta) [inline spring-xd-modules.tex]: operations (such as count and gauge) on the

File: spring-xd-modules.tex [hunk 8]
```diff
@@ -72,20 +74,19 @@ \subsection{Processor}
...
+incoming data and store the result into metric repository(See section~\ref{sec:Analytics}.)
```
Comment (pperalta) [inline spring-xd-modules.tex]: into a metric repository

File: spring-xd-modules.tex [hunk 9]
```diff
@@ -72,20 +74,19 @@ \subsection{Processor}
...
 \par
 
-The basic sink is comprised of a ``input'' channel and a bean.
-The input channel receives all messages from the stream and dispatches
-messages to the bean (see figure~\ref{fig:sinkmbc}.) It is the responsibility 
-of the bean to implement retry behavior in cases of failure. The sinks 
-included in Spring XD have configurable options for retries.
+The basic sink is comprised of a ``input'' channel connector and a data processing
+component. The input channel receives data from the stream and dispatches
+messages to the bean (see figure~\ref{fig:sinkmbc}.) The sinks
```
Comment (pperalta) [inline spring-xd-modules.tex]: Are we still using the term "bean"?

File: spring-xd-modules.tex [hunk 10]
```diff
@@ -99,16 +100,19 @@ \subsection{Sink}
 \par
 
 \subsection{Job}
-Spring XD uses Spring Batch \cite{spring-batch-reference} as the foundation for implementing
+Spring XD uses Spring Batch \cite{spring-batch-reference}, a JSR standardization (JSR-352)
```
Comment (pperalta) [inline spring-xd-modules.tex]: JSR standard

File: spring-xd-modules.tex [hunk 11]
```diff
@@ -99,16 +100,19 @@ \subsection{Sink}
...
+of batch workload data processing as the foundation for implementing
```
Comment (pperalta) [inline spring-xd-modules.tex]: for batch workload

File: spring-xd-modules.tex (hunk 9 above)
Comment (ilayaperumalg) [inline spring-xd-modules.tex]: no, sorry, missed this one :-)

File: spring-xd-modules.tex [hunk 12]
```diff
@@ -122,20 +126,23 @@ \subsection{Job}
...
-\texttt{modules/sink} and \texttt{modules/job}.
-These modules are dynamically loaded when a stream or job is deployed that requires that
-specific module.
-
-\par
 \subsection{Composite}
-A Composite module provides a way to create a single module 
-comprised of other modules (processing chain).  A composite module can be used to prevent 
-duplication when a processing chain of modules is used frequently.  
+A Composite module provides a way to create a single module comprised of
```
Comment (pperalta) [inline spring-xd-modules.tex]: s/Composite/composite

**Comment by pperalta** [comment]
beb921062448247a522d61178f91f624422621bf

**Pull request closed without merging**